versioned_docs/version-5.x/configuring-links.md

```diff
@@ -0,0 +1,355 @@
+
+```js
+{
+  routes: [
+    {
+      name: 'Chat',
+      params: {
+        sort: 'latest',
+      },
+    },
+  ];
+}
+```
+
+You can specify a separate `config` option to control how the deep link is parsed to suit your needs.
+
+```js
+const linking = {
+  prefixes: ['https://mychat.com', 'mychat://'],
+  config: {
+    Chat: 'feed/:sort',
+  },
+};
+```
+
+Here `Chat` is the name of the screen that handles this URL. The navigator will need to have a `Chat` screen which handles a `sort` param for the route:
+
+You can also customize how params are parsed, for example, if you parse the path `/item/42` as `item/:id`, the param `id` will be parsed as string by default. But you can customize it by passing a function:
+
+```js
+{
+  Catalog: {
+    path: 'item/:id',
+    parse: {
+      id: Number,
+    },
+  },
+}
+```
+
+This will result in something like:
+
+```js
+const state = {
+  routes: [
+    {
+      name: 'Catalog',
+      params: { id: 42 },
+    },
+  ],
+};
+```
+
+It's important to note that the state object must match the hierarchy of nested navigators. Otherwise the state will be discarded.
+
+Sometimes we'll have the target navigator nested in other navigators which aren't part of the deep link. For example, let's say our navigation structure looks like this:
+
+```js
+function Home() {
+  return (
+    <Tab.Navigator>
+      <Tab.Screen name="Profile" component={Profile} />
+      <Tab.Screen name="Notifications" component={Notifications} />
+    </Tab.Navigator>
+  );
+}
+
+function App() {
+  return (
+    <Stack.Navigator>
+      <Stack.Screen name="Home" component={Home} />
+      <Stack.Screen name="Settings" component={Settings} />
+    </Stack.Navigator>
+  );
+}
+```
+
+Here we have a stack navigator in the root, and inside the `Home` screen of the root stack, we have a tab navigator with various screens. With this structure, let's say we want the path `/users/:id` to go to the `Profile` screen. We can express the nested config like so:
+
+```js
+{
+  Home: {
+    screens: {
+      Profile: 'users/:id',
+    },
+  },
+}
+```
+
+In this config, we specify that the `Profile` screen should be resolved for the `users/:id` pattern and it's nested inside the `Home` screen. Then parsing `users/jane` will result in the following state object:
+
+```js
+const state = {
+  routes: [
+    {
+      name: 'Home',
+      state: {
+        routes: [
+          {
+            name: 'Profile',
+            params: { id: 'jane' },
+          },
+        ],
+      },
+    },
+  ],
+};
+```
+
+Sometimes we want to ensure that a certain screen will always be present for the navigator in the state object. We can achieve it by specifying the `initialRouteName` property for that navigator in the config. For the above example, if we want the `Notifications` screen to be the initial route in `Home` tab navigator, we should specify such config (`Notifications` screen needn't be mentioned in `screens` property):
```

Comment:
the last bit in parenthesis here is a little confusing - do you mean that you don't need to include it in the screens property if you don't want to customize the path?

Reply:
Most of the deep linking doc was written by @WoLewicki, I just moved it here, so maybe he can clarify

Reply:
Yes @brentvatne it is what I meant, it can be included in `initialRouteName` but can be not customized, so not mentioned in `screens`. Maybe it is a bit confusing and can be deleted if you think so.